src/main/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilder.java

```diff
@@ -453,7 +453,7 @@ else if (STREAM_TYPES.contains(this.outType))
 
         switch (typeToBuild) {
             case UNSPECIFIED:
-                throw new IllegalArgumentException("Must specify file or stream output type.");
+                throw new IllegalArgumentException("Output format type is not set, or could not be inferred from the output path. If a path was used, does it have a valid VCF extension?");
```

Comment:
if you're at it, why not take htsjdk.samtools.util.IOUtil#VCF_EXTENSIONS_LIST and present it to the user for reference?